Data/Boltzmann/Internal/Parser.hs

```diff
@@ -107,8 +107,7 @@ parseFromFile
 parseFromFile p file = runParser p file <$> readFile file
 
 -- | Prints the given parsing errors.
-printError :: (Stream t, ShowErrorComponent e) => ParseErrorBundle t e -> IO a
-
+printError :: (ShowErrorComponent e, VisualStream t, TraversableStream t) => ParseErrorBundle t e -> IO a
```

Comment:
Why are these type changes required?

Reply:
I mean not just here, but throughout the PR.

Reply:
Needed them to make it compile when I ran cabal build.

Reply:
It seems that these type changes are imposed by an implicit `megaparsec` update to version `9.0.0` or higher (see https://hackage.haskell.org/package/megaparsec-9.0.1/docs/Text-Megaparsec-Stream.html#t:VisualStream). This version of `megaparsec` is not even in the latest LTS resolver of stack, (see https://www.stackage.org/lts-16.27). How have you managed to build `boltzmann-brain` while keeping the `haskell-ext-src` untouched?

Reply:
I used nixos-20.09 as a package set and used `cabal build` in the nix shell. Then I followed the ghc errors and to get compilation working. As mentioned below I rolled back the package set to nixos-20.03 which has the megaparsec version you want. BTW, nix picks up its haskell packages from stack, although strict matching is not guaranteed.